src/utils.ts

```diff
@@ -1405,3 +1428,32 @@ export function parsePackageVersion({ version }: { version: string }): {
   const [major, minor, patch] = version.split('.').map((n: string) => Number.parseInt(n, 10));
   return { major, minor, patch };
 }
+
+/**
+ * Fisher–Yates Shuffle
+ *
+ * Reference: https://bost.ocks.org/mike/shuffle/
+ * @param sequence - items to be shuffled
+ * @param limit - the number of ite
+ */
+export function shuffle<T>(sequence: Iterable<T>, limit = 0): Array<T> {
+  const items = Array.from(sequence); // shallow copy in order to never shuffle the input
+
+  if (limit > items.length) {
+    throw new MongoInvalidArgumentError('Limit must be less than the number of items');
```

Comment:
There isn't a good specific one, I think staying general is fine here since `ShuffleError` would be too generic, and `SrvShuffleError` would be too specific.